src/event-bridge/event-bus.ts

```diff
@@ -258,19 +302,65 @@ abstract class EventBusBase<E extends EventBusRuleInput>
   /**
    * @inheritdoc
    */
+  when<O extends E>(
+    id: string,
+    predicate: EventPredicateFunction<E, O>
+  ): Rule<E, O>;
   when<O extends E>(
     scope: Construct,
     id: string,
     predicate: EventPredicateFunction<E, O>
-  ): EventBusRule<E, O> {
-    return new EventBusRule<E, O>(scope, id, this as any, predicate);
+  ): Rule<E, O>;
+  when<O extends E>(
+    scope: Construct | string,
+    id?: string | EventPredicateFunction<E, O>,
+    predicate?: EventPredicateFunction<E, O>
+  ): Rule<E, O> {
+    if (predicate) {
+      return new Rule<E, O>(
+        scope as Construct,
+        id as string,
+        this as any,
+        predicate
+      );
+    } else {
+      return new Rule<E, O>(
+        this.bus,
+        scope as string,
+        this as any,
+        id as EventPredicateFunction<E, O>
+      );
+    }
+  }
+
+  /**
+   * @inheritdoc
```

Comment:
Is this necessary?

Reply:
Guess not, I didn't think it would pull the docs in on its own, I guess it does.